src/ducks/Signals/signalFormManager/signalMaster/SignalMaster.js

```diff
@@ -1,183 +1,161 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
 import { push } from 'react-router-redux'
 import cx from 'classnames'
-import { graphql } from 'react-apollo'
 import { compose } from 'recompose'
 import { connect } from 'react-redux'
 import { createTrigger, updateTrigger } from '../../common/actions'
 import Message from '@santiment-network/ui/Message'
 import Toggle from '@santiment-network/ui/Toggle'
 import TriggersForm from '../signalCrudForm/signalsList/TriggersForm'
 import AboutForm from '../aboutForm/AboutForm'
-import { TRIGGER_BY_ID_QUERY } from '../../common/queries'
-import {
-  mapTriggerToFormProps,
-  mapFormPropsToTrigger,
-  mapGQLTriggerToProps
-} from '../../utils/utils'
+import { mapTriggerToFormProps, mapFormPropsToTrigger } from '../../utils/utils'
 import { SIGNAL_ROUTES, TRIGGER_STEPS } from '../../common/constants'
 import styles from '../signalCrudForm/signal/TriggerForm.module.scss'
 
-export class SignalMaster extends React.PureComponent {
-  static defaultProps = {
-    canRedirect: true,
-    step: TRIGGER_STEPS.SETTINGS
-  }
+const getTitle = (step, formData, { id }, isShared) => {
+  const isUpdate = id > 0 && !isShared
 
-  state = {
-    step: TRIGGER_STEPS.SETTINGS,
-    trigger: {
-      title: `Signal_[${new Date().toLocaleDateString('en-US')}]`,
-      description: '',
-      isActive: true,
-      isPublic: false
+  switch (step) {
+    case TRIGGER_STEPS.SETTINGS: {
+      return isUpdate ? 'Update signal' : 'Create signal'
     }
-  }
-
-  componentDidUpdate (prevProps) {
-    const { trigger } = this.state
-
-    const newProps = this.props
-    if (newProps.trigger && newProps.trigger.trigger && !trigger.id) {
-      this.setState({
-        trigger: {
-          ...newProps.trigger.trigger
-        },
-        step: newProps.step
-      })
+    case TRIGGER_STEPS.CONFIRM: {
+      if (isUpdate) {
+        return formData.isPublic
+          ? 'Update public signal'
+          : 'Update private signal'
```

Comment:
to reduce repetitions - move `formData.isPublic ? 'public' : 'private'` to separate var